securedrop_client/logic.py

```diff
@@ -594,17 +594,17 @@ def on_reply_download_failure(self, exception: Exception) -> None:
 
     def downloaded_file_exists(self, file_uuid: str) -> bool:
         '''
-        Check if the file specified by file_uuid exists. If it doesn't sync the api so that any
-        missing files, including this one, are updated to be re-downloaded.
+        Check if the file specified by file_uuid exists. If it doesn't update the local db and
+        GUI to show the file as not downloaded.
         '''
         file = self.get_file(file_uuid)
         fn_no_ext, dummy = os.path.splitext(os.path.splitext(file.filename)[0])
         filepath = os.path.join(self.data_dir, fn_no_ext)
         if not os.path.exists(filepath):
-            self.gui.update_error_status(_(
-                'File does not exist in the data directory. Please try re-downloading.'))
             logger.debug('Cannot find {} in the data directory. File does not exist.'.format(
                 file.original_filename))
+            storage.update_missing_files(self.data_dir, self.session)
+            self.session.refresh(file)
```

Comment:
I don't think this updates the UI until the conversation view is redrawn for other reasons, i.e. if we click _off_ this source and back onto the source corresponding to the missing file or if a sync happens to finish (btw to quickly see what parts of the UI are updating naturally even without network I've been commenting out [this adding of the metadata sync job](https://github.com/freedomofpress/securedrop-client/blob/master/securedrop_client/logic.py#L397)). 

I think we want to make sure that the state of the UI is representing the state of the locally stored data even without frequent network syncs, since it's expected that even in the best conditions there will be multi-second delay, let me know what you think

Reply:
so i removed the refresh line since it is unnecessary.

also, just an explanation of what i'm trying to do in this pr:  instead of relying on sync_api, we now call update_missing_files directly when we learn that the file does not exist. this made it so we could remove 3 unnecessary sync_api calls and you can see that it still works:

![missing-file](https://user-images.githubusercontent.com/4522213/73301446-a58d0000-41c7-11ea-8f76-1923de42e35b.gif)

Reply:
oops that gif was with the `self.session.refresh(file)` line of code, which i removed. here's a gif without `self.session.refresh(file)`:

![without-refresh](https://user-images.githubusercontent.com/4522213/73301570-dd944300-41c7-11ea-944a-a020214a436c.gif)

Reply:
ah, there was a call to sync_api from the GUI... that explains why this was working. since we don't want calls to sync_api, i updated the GUI to receive a file_missing signal from the Controller instead